core/rio/hdt/src/main/java/org/eclipse/rdf4j/rio/hdt/part/HDTDictionarySectionFactory.java

```diff
@@ -0,0 +1,26 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.rio.hdt.part;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * HDT DictionarySection factory.
+ * 
+ * @author Bart Hanssens
+ */
+public class HDTDictionarySectionFactory {
+	public static HDTDictionarySection parse(InputStream is) throws IOException {
+		int dtype = is.read();
+		if (dtype != HDTDictionarySection.Type.FRONT.getValue()) {
+			throw new UnsupportedOperationException("Dictionary section: only front encoding is supported");
```

Comment:
It would be useful to add to this message the `dtype` value that was found (for debugging purposes)

Reply:
good idea, added